packages/next/server/config.ts

```diff
@@ -138,7 +140,29 @@ function assignDefaults(userConfig: { [key: string]: any }) {
     {}
   )
 
-  const result = { ...defaultConfig, ...config }
+  // Copy properties we intend to mutate.
+  const copy = (nextConfig: NextConfig): NextConfig => {
```

Comment:
This seems prone to error in a new prop is added to next.config but then missing from here.

Perhaps a more general deepClone() is necessary.

Reply:
I'd agree, but didn't see a deepClone function in the codebase, so instead I just targeted the properties that were mutated in the function below this line.